core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -0,0 +1,315 @@
+import static org.apache.iceberg.TableProperties.COMMIT_MIN_RETRY_WAIT_MS_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_NUM_RETRIES_DEFAULT;
+import static org.apache.iceberg.TableProperties.COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT;
+
+public class CatalogHandlers {
+  private static final Schema EMPTY_SCHEMA = new Schema();
+
+  private CatalogHandlers() {
+  }
+
+  static class UnprocessableEntityException extends RuntimeException {
+    UnprocessableEntityException(Set<String> commonKeys) {
+      super(String.format("Invalid namespace update, cannot set and remove keys: %s", commonKeys));
+    }
+  }
+
+  private static class ValidationFailureException extends RuntimeException {
+    private final CommitFailedException wrapped;
+
+    private ValidationFailureException(CommitFailedException cause) {
+      super(cause);
+      this.wrapped = cause;
+    }
+
+    public CommitFailedException wrapped() {
+      return wrapped;
+    }
+  }
+
+  public static ListNamespacesResponse listNamespaces(SupportsNamespaces catalog, Namespace parent) {
+    List<Namespace> results;
+    if (parent.isEmpty()) {
+      results = catalog.listNamespaces();
+    } else {
+      results = catalog.listNamespaces(parent);
+    }
+
+    return ListNamespacesResponse.builder().addAll(results).build();
+  }
+
+  public static CreateNamespaceResponse createNamespace(SupportsNamespaces catalog, CreateNamespaceRequest request) {
+    Namespace namespace = request.namespace();
+    catalog.createNamespace(namespace, request.properties());
+    return CreateNamespaceResponse.builder()
+        .withNamespace(namespace)
+        .setProperties(catalog.loadNamespaceMetadata(namespace))
+        .build();
+  }
+
+  public static GetNamespaceResponse loadNamespace(SupportsNamespaces catalog, Namespace namespace) {
+    Map<String, String> properties = catalog.loadNamespaceMetadata(namespace);
+    return GetNamespaceResponse.builder()
+        .withNamespace(namespace)
+        .setProperties(properties)
+        .build();
+  }
+
+  public static DropNamespaceResponse dropNamespace(SupportsNamespaces catalog, Namespace namespace) {
+    boolean dropped = catalog.dropNamespace(namespace);
+    return DropNamespaceResponse.builder()
+        .dropped(dropped)
+        .build();
+  }
+
+  public static UpdateNamespacePropertiesResponse updateNamespaceProperties(
+      SupportsNamespaces catalog, Namespace namespace, UpdateNamespacePropertiesRequest request) {
+    Set<String> removals = Sets.newHashSet(request.removals());
+    Map<String, String> updates = request.updates();
+
+    Set<String> commonKeys = Sets.intersection(updates.keySet(), removals);
+    if (!commonKeys.isEmpty()) {
+      throw new UnprocessableEntityException(commonKeys);
+    }
+
+    Map<String, String> startProperties = catalog.loadNamespaceMetadata(namespace);
+    Set<String> missing = Sets.difference(removals, startProperties.keySet());
+
```

Comment:
Another potential validation here, we are allowed to request the removal of a property that is not set, is that ok? Seems fine to me since I think a lot of property structures let you remove a non-existing key without error.

Reply:
The idea is for these to be idempotent, so it is fine to request removing a property that is not set.

If we didn't allow that, then you'd get incorrect errors when a user does something like `SHOW TBLPROPERTIES` followed by concurrent `UNSET` commands. Retries would also cause problems.